### ISSUE TITLE
deepvac_human_accept600 SOTA

### ISSUE THREAD

**Issue by gemfield** [DeepVAC/ESPNet#15]
# IoU 金榜
- test9: 91.5%
- test5: 90.9%
- test6: 90.5% 
- test7: 90.5%
- test3: 87.2%
- test2: 86.9%
- test4: 86.3%
- test1: 85.7%
- test8: 85.6%

**Comment by gemfield** [comment]
# test1
- 训练集：clothes std 2.1
- config.input_w = 384
- config.input_h = 384
- config.core.cls_num = 4
- config.aug.ImageWithMasksRandomRotateAug.max_angle = 45
- config.core.batch_size = 8
- config.core.mean = config.data['mean']
- config.core.std = config.data['std']
- config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
- config.core.optimizer = torch.optim.SGD(config.core.net.parameters(), 5e-3, momentum=0.9)
- config.core.scheduler = optim.lr_scheduler.MultiStepLR(config.core.optimizer, milestones=[20,40,55,70,80,90,100,110,120,130,140,150,160], gamma=0.27030)
- config.core.criterion = torch.nn.CrossEntropyLoss(weight)
- AugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 =>UDmotionAug@0.1 => NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.15 => BrightnessJitterAug@0.15 =>ContrastJitterAug@0.15 => ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksNormalizeAug =>ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>ImageWithMasksToTensorAug', deepvac_config)

**Comment by 1icas** [comment]
# test1
## IoU

- label: 85.7%
- label_no_mix: ??

## 特别超参

- 训练集：HumanSeg-Mapillary
- config.input_w = 224
- config.input_h = 384
- config.core.optimizer = torch.optim.SGD(config.core.net.parameters(), 0.01, momentum=0.9)
-  config.core.scheduler = optim.lr_scheduler.MultiStepLR(config.core.optimizer, milestones=[4, 8, 12, 15], gamma=0.1)

## 基本超参
    config.core.cls_num = 2
    config.core.batch_size = 64
    config.core.mean = config.data['mean']
    config.core.std = config.data['std']
    config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
    config.core.criterion = torch.nn.CrossEntropyLoss(weight)
    AugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 =>UDmotionAug@0.1 => NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.15 => BrightnessJitterAug@0.15 =>ContrastJitterAug@0.15 => ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksNormalizeAug =>ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>ImageWithMasksToTensorAug', deepvac_config)

**Comment by 1icas** [comment]
# test2
## IoU

- label: 86.9% 
- label_no_mix: ??

## 特别超参

- 训练集：HumanSeg-Mapillary + 1倍HumanSeg-Mapillary 背景替换
- config.input_w = 224
- config.input_h = 384
- config.core.optimizer = torch.optim.SGD(config.core.net.parameters(), 0.01, momentum=0.9)
-  config.core.scheduler = optim.lr_scheduler.MultiStepLR(config.core.optimizer, milestones=[4, 8, 12, 15], gamma=0.1)

## 基本超参
      config.core.cls_num = 2
      config.core.batch_size = 64
      config.core.mean = config.data['mean']
      config.core.std = config.data['std']
      config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
      config.core.criterion = torch.nn.CrossEntropyLoss(weight)
      AugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 =>UDmotionAug@0.1 => NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.15 => BrightnessJitterAug@0.15 =>ContrastJitterAug@0.15 => ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksNormalizeAug =>ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>ImageWithMasksToTensorAug', deepvac_config)

**Comment by 1icas** [comment]
# test3
## IoU

- label: 87.2%
- label_no_mix: ??

## 特别超参

- 训练集：HumanSeg-Mapillary + 4倍HumanSeg-Mapillary 背景替换
- config.input_w = 224
- config.input_h = 384
- config.core.optimizer = torch.optim.SGD(config.core.net.parameters(), 0.01, momentum=0.9)
-  config.core.scheduler = optim.lr_scheduler.MultiStepLR(config.core.optimizer, milestones=[4, 8, 12, 15], gamma=0.1)


## 基本超参
      config.core.cls_num = 2
      config.core.batch_size = 64
      config.core.mean = config.data['mean']
      config.core.std = config.data['std']
      config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
      config.core.criterion = torch.nn.CrossEntropyLoss(weight)
      AugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 =>UDmotionAug@0.1 => NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.15 => BrightnessJitterAug@0.15 =>ContrastJitterAug@0.15 => ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksNormalizeAug =>ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>ImageWithMasksToTensorAug', deepvac_config)

**Comment by 1icas** [comment]
# test4
## IoU

- label: 86.3%
- label_no_mix: ??
  
## 特别超参

- 训练集：HumanSeg-Mapillary + 4倍HumanSeg-Mapillary 背景替换 + 2w张纯背景
- config.input_w = 224
- config.input_h = 384
- config.core.optimizer = torch.optim.SGD(config.core.net.parameters(), 0.01, momentum=0.9)
-  config.core.scheduler = optim.lr_scheduler.MultiStepLR(config.core.optimizer, milestones=[4, 8, 12, 15], gamma=0.1)

## 基本超参

      config.core.cls_num = 2
      config.core.batch_size = 64
      config.core.mean = config.data['mean']
      config.core.std = config.data['std']
      config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
      config.core.criterion = torch.nn.CrossEntropyLoss(weight)
      AugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 =>UDmotionAug@0.1 => NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.15 => BrightnessJitterAug@0.15 =>ContrastJitterAug@0.15 => ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksNormalizeAug =>ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>ImageWithMasksToTensorAug', deepvac_config)

**Comment by 1icas** [comment]
# test5
## IoU

- label: 90.9%
- label_no_mix:90.8%

## 特别超参

- 训练集：HumanSeg-Mapillary + CoCo + 2.5w张纯背景
- config.core.optimizer = torch.optim.SGD(config.core.net.parameters(), 0.01, momentum=0.9, weight_decay=1e-5) 
-  config.core.scheduler = optim.lr_scheduler.MultiStepLR(config.core.optimizer, milestones=[6, 12, 15, 18], gamma=0.1) 

## 模型修改
- 去掉了网络中所有的nn.Dropout2d

## 基本超参
    config.input_w = 384
    config.input_h = 384
    config.core.cls_num = 2
    config.core.batch_size = 16
    config.core.mean = config.data['mean']
    config.core.std = config.data['std']
    config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
    config.core.criterion = torch.nn.CrossEntropyLoss(weight)
    AugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 =>UDmotionAug@0.1 => NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.15 => BrightnessJitterAug@0.15 =>ContrastJitterAug@0.15 => ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksNormalizeAug =>ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>ImageWithMasksToTensorAug', deepvac_config)

**Comment by MHGL** [comment]
# **test6**
## IoU
- epoch: 93
- labels: 90.5%
- labels_no_mix: 90.1%
## 特别超参
- 训练集： deepvac-chengdu(1526 no mix) + deepvac-waibao(1589 no mix)
- config.input_w = 384
- config.input_h = 384
- config.core.criterion = torch.nn.CrossEntropyLoss(weight)
## 基本超参
```python
config.core.epoch_num = 100
config.core.cls_num = 2
config.core.batch_size = 16
config.core.mean = config.data['mean']
config.core.std = config.data['std']
config.core.optimizer = torch.optim.Adam(config.core.net.parameters(), 3e-4, (0.9, 0.999), eps=1e-08, weight_decay=5e-4)
lambda_lr = lambda epoch: round ((1 - epoch/config.core.epoch_num) ** 0.9, 8)
config.core.scheduler = optim.lr_scheduler.LambdaLR(config.core.optimizer, lr_lambda=lambda_lr)
config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
AugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 =>UDmotionAug@0.1 => NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.15 => BrightnessJitterAug@0.15 =>ContrastJitterAug@0.15 => ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksNormalizeAug =>ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>ImageWithMasksToTensorAug', deepvac_config)
```

**Comment by MHGL** [comment]
# **test7**
## IoU
- epoch: 95
- labels: 90.5%
- labels_no_mix: 90.3%
## 特别超参
- 训练集： deepvac-chengdu(1526 no mix) + deepvac-waibao(1589 no mix)
- config.input_w = 384
- config.input_h = 384
- config.core.criterion = bce_loss + dice_loss 
## 基本超参
```python
config.core.epoch_num = 100
config.core.cls_num = 2
config.core.batch_size = 16
config.core.mean = config.data['mean']
config.core.std = config.data['std']
config.core.optimizer = torch.optim.Adam(config.core.net.parameters(), 3e-4, (0.9, 0.999), eps=1e-08, weight_decay=5e-4)
lambda_lr = lambda epoch: round ((1 - epoch/config.core.epoch_num) ** 0.9, 8)
config.core.scheduler = optim.lr_scheduler.LambdaLR(config.core.optimizer, lr_lambda=lambda_lr)
config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
AugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 =>UDmotionAug@0.1 => NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.15 => BrightnessJitterAug@0.15 =>ContrastJitterAug@0.15 => ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksNormalizeAug =>ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>ImageWithMasksToTensorAug', deepvac_config)
```

**Comment by MHGL** [comment]
# **test8**
## IoU
- 前sota模型
- labels: 85.6%
- labels_no_mix: 85.5%
## 特别超参
- 训练集： deepvac-chengdu(1526 no mix) + supervisely(5711)
- config.input_w = 384
- config.input_h = 384
- config.core.criterion = bce_loss + dice_loss 
## 基本超参
```python
config.core.epoch_num = 100
config.core.cls_num = 2
config.core.batch_size = 16
config.core.mean = config.data['mean']
config.core.std = config.data['std']
config.core.optimizer = torch.optim.Adam(config.core.net.parameters(), 3e-4, (0.9, 0.999), eps=1e-08, weight_decay=5e-4)
lambda_lr = lambda epoch: round ((1 - epoch/config.core.epoch_num) ** 0.9, 8)
config.core.scheduler = optim.lr_scheduler.LambdaLR(config.core.optimizer, lr_lambda=lambda_lr)
config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
```

**Comment by MHGL** [comment]
# **test9**
## IoU
- epoch: 71
- labels: 91.5%
- labels_no_mix: 91.4%
## 特别超参
- 训练集： deepvac-chengdu(1646 no mix) + deepvac-waibao(4399 no mix) + onlyBG(1000)
- config.input_w = 384
- config.input_h = 384
- config.core.criterion = CrossEntropyLoss
## 基本超参
```python
config.core.epoch_num = 100
config.core.cls_num = 2
config.core.batch_size = 16
config.core.mean = config.data['mean']
config.core.std = config.data['std']
config.core.optimizer = torch.optim.Adam(config.core.net.parameters(), 3e-4, (0.9, 0.999), eps=1e-08, weight_decay=5e-4)
lambda_lr = lambda epoch: round ((1 - epoch/config.core.epoch_num) ** 0.9, 8)
config.core.scheduler = optim.lr_scheduler.LambdaLR(config.core.optimizer, lr_lambda=lambda_lr)
config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
AugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 =>UDmotionAug@0.1 => NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.15 => BrightnessJitterAug@0.15 =>ContrastJitterAug@0.15 => ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksNormalizeAug =>ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>ImageWithMasksToTensorAug', deepvac_config)
```
